lib/model/user_model.dart

```diff
@@ -0,0 +1,112 @@
+// ignore_for_file: always_put_required_named_parameters_first
+class UserState {
+  UserState({
+    required this.isLoading,
+    this.user,
+    // Placeholder for future helpers (e.g. error handling)
+    required this.errorMessage,
+  });
+
+  // Default constructor
+  UserState.defaultConstructor()
+      : isLoading = false,
+        errorMessage = '',
+        user = null;
+
+  bool isLoading;
+  User? user;
+  String errorMessage;
+
+  void setUser(User newUser) {
+    print("User set to: ${newUser.name}");
+    user = newUser;
+  }
+
+  void removeUser() {
+    user = null;
+  }
+
+  void setIsLoading(bool isLoading) {
+    this.isLoading = isLoading;
+  }
+
+  void toggleIsLoading() {
+    isLoading = !isLoading;
+  }
+}
+
+class User {
+  User({
+    required this.id,
+    required this.name,
+    required this.lastName,
+    this.email = '',
+    this.matriculationNumber = '',
+    this.lrzId = '',
+    this.role = 3,
+    this.password = '',
+    required this.courses,
+    required this.administeredCourses,
+    required this.pinnedCourses,
+    required this.settings,
+    required this.bookmarks,
+  });
+
+  int id;
+  String name;
+  String lastName;
+  String email;
+  String matriculationNumber;
+  String lrzId;
+  int role;
+  String password;
+  List<Course> courses;
+  List<Course> administeredCourses;
+  List<Course> pinnedCourses;
+  List<UserSetting> settings;
+  List<Bookmark> bookmarks;
+}
+
+class Course {
+  Course({
+    required this.id,
+    required this.name,
+  });
+
+  int id;
+  String name;
+}
```

Comment:
The file's name is user_model, but it also has the Course model? For me, it could make sense if it gathers all the models related to the User, but what was the reason for you to put them all here?

Reply:
Initially this was only meant to be a placeholder (depending on what models would then be needed), but of course it makes more sense to have the models in different files and grouped by category (e.g., user, course, ...).
Should be fixed in the latest commit :wink:.